### PR TITLE
fix(html): elgg_normalize_url() handles tel: links

### LINKED ISSUE #10689: elgg_normalize_url should not add site url to tel:
as tel: links are valid html5 links, elgg_normalize_url should not add site url

### PR DESCRIPTION
**Pull request by mrclay**
Removes cruft for older PHP versions.

Fixes #10689

### CODE REVIEW

**Review by hypeJunction — APPROVED** [review]

File: engine/lib/output.php [hunk 1]
```diff
@@ -288,53 +288,38 @@ function elgg_format_element($tag_name, array $attributes = array(), $text = '',
...
-	} else {
-		$validated = filter_var($url, FILTER_VALIDATE_URL);
+	if (filter_var($url, FILTER_VALIDATE_URL)) {
+		return $url;
 	}
 
-	// work around for handling absoluate IRIs (RFC 3987) - see #4190
-	if (!$validated && (strpos($url, 'http:') === 0) || (strpos($url, 'https:') === 0)) {
-		$validated = true;
+	if (preg_match("#^([a-z]+)\\:#", $url, $m)) {
```
Comment (jdalsem) [inline engine/lib/output.php]: we could also do a `parse_urls` and check for the scheme key

**Comment by jdalsem** [comment]
This also fixes #7178

**Comment by jdalsem** [comment]
PR is fine, but should be on 2.3

**Comment by mrclay** [comment]
Moved to 2.3

**Comment by mrclay** [comment]
I don't understand the test failure. Maybe Travis isn't re-running against the new branch? I'll re-open

**Comment by mrclay** [comment]
Replaced by #10694

**Pull request closed without merging**